server/services/alert.js

```diff
@@ -1,39 +1,103 @@
+    .then(group => generateAllPageRequests(group, ticket))
+    .then(pageRequests => sendPageRequestsToQueue(pageRequests))
     .then(() => ticketService.addAction(ticket.id, actionTypes.CREATED));
 }
 
-function sendPage(ticket, user, device) {
-  ticketService.addAction(ticket.id, actionTypes.PAGE_SENT, user.id, device);
-
-  switch (device.type) {
-    case ('email'):
-      return emailService.sendEmail(ticket, user, device);
-    case ('sms'):
-      return phoneService.sendMessage(ticket, user, device);
-    case ('phone'):
-      return phoneService.makeCall(ticket, user, device);
-    default:
-      return Promise.reject(new Error(`Invalid device type: ${device.type} on User: "${user.id}"`));
+function generateAllPageRequests(group, ticket) {
+  let pageRequests = [];
+  let currentDelay = 0;
+  const title = ticket.metadata.title;
+
+  for (const user of group.escalationPolicy.subscribers) {
+    const userPages = generateUserPageRequests(ticket.id, user, currentDelay, title);
+    pageRequests = pageRequests.concat(userPages);
+    currentDelay += group.escalationPolicy.pagingIntervalInMinutes;
+  }
+  return pageRequests;
+}
+
+function generateUserPageRequests(ticketId, user, delay, title) {
+  const userPageRequests = [];
+  let currUserDelay = 0;
+
+  for (let i = 0; i < user.devices.length; i++) {
+    if (i !== 0) {
+      currUserDelay += user.delays[i - 1] || config.defaultDelay;
+    }
+
+    userPageRequests.push({
+      ticketId,
+      userId: user.id,
+      device: user.devices[i],
+      delay: (delay + currUserDelay) * ONE_MINUTE_MILLISECONDS,
+      title
+    });
   }
+  return userPageRequests;
+}
+
+function sendPageRequestsToQueue(userPageRequests) {
+  const options = {
+    method: 'POST',
+    uri: `${config.queueHost}/${config.queuePath}`,
+    body: {
+      pages: userPageRequests
+    },
+    json: true
+  };
+
+  return request(options)
+    .then((futurePages) => {
+      if (!futurePages.length) {
+        return;
+      }
+      const ticketId = futurePages[0].data.ticketId;
+      const pageIds = futurePages.map(page => page.id);
+      ticketService.updateTicket(ticketId, { pageIds });
+    });
+}
+
+function sendPage(ticketId, userId, device) {
+  ticketService.addAction(ticketId, actionTypes.PAGE_SENT, userId, device);
+
+  const userPromise = userService.getUser(userId);
+  const ticketPromise = ticketService.getById(ticketId);
+
+  return Promise.all([userPromise, ticketPromise])
+    .then((results) => {
+      const user = results[0];
+      const ticket = results[1];
+
+      switch (device.type) {
+        case ('email'):
+          return emailService.sendEmail(ticket, user, device);
+        case ('sms'):
+          return phoneService.sendMessage(ticket, user, device);
+        case ('phone'):
+          return phoneService.makeCall(ticket, user, device);
+        default:
+          return Promise.reject(new Error(`Invalid device type: ${device.type} on User: "${user.id}"`));
```

Comment:
I'm iffy on the logic here. We add the action to the ticket on line 76 before checking the type of the device. If the device type doesn't exist, we're clogging up the ticket actions array with incorrect actions.